soh/src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -6121,13 +6131,15 @@ s32 func_8083E5A8(Player* this, GlobalContext* globalCtx) {
                 iREG(67) = false;
 
                 if (gSaveContext.n64ddFlag && giEntry.getItemId == RG_ICE_TRAP) {
-                    this->stateFlags1 &= ~(PLAYER_STATE1_10 | PLAYER_STATE1_11);
-                    this->actor.colChkInfo.damage = 0;
-                    func_80837C0C(globalCtx, this, 3, 0.0f, 0.0f, 0, 20);
-                    Player_SetPendingFlag(this, globalCtx);
-                    this->getItemId == GI_NONE;
-                    this->getItemEntry = (GetItemEntry)GET_ITEM_NONE;
-                    return 1;
+                    if (giEntry.getItemFrom == ITEM_FROM_FREESTANDING) {
+                        this->stateFlags1 &= ~(PLAYER_STATE1_10 | PLAYER_STATE1_11);
+                        this->actor.colChkInfo.damage = 0;
+                        func_80837C0C(globalCtx, this, 3, 0.0f, 0.0f, 0, 20);
+                        Player_SetPendingFlag(this, globalCtx);
+                        this->getItemId = GI_NONE;
+                        this->getItemEntry = (GetItemEntry) GET_ITEM_NONE;
```

Comment:
i know it's not quite identical, but maybe there's a way to move some of this code to a reusable location? (not something that needs to happen for this pr, just a note for possible future code cleanup)